web/geolocation/geolocation.js

```diff
@@ -0,0 +1,521 @@
+                    var ul = document.getElementById("history-list");
+                    ul.appendChild(div);
+                }
+                redraw(payload);
+            }
+            catch(e){
+                console.log(e);
+            }
+        },
+        objects: async objectEvent => {
+            if (
+                objectEvent.message.type == 'uuid' &&
+                objectEvent.message.event == 'delete' &&
+                objectEvent.message.data.id == pubnub.getUUID()
+            ) {
+                //  The Object associated with OUR UUID was deleted.
+                //  log out.  This could have been caused e.g. by a duplicate tab logging out
+                location.href = '../index.html'
+            }
+            else if (
+                objectEvent.message.type == 'uuid' &&
+                objectEvent.message.event == 'delete' &&
+                objectEvent.message.data.id != pubnub.getUUID()
+            ) {
+                var userId = objectEvent.message.data.id
+                //  Remove member from geolocation channel
+                if (channelMembers.hasOwnProperty(userId)) {
+                    removeUserFromGeoChannel(userId)
+                }
+            }
+            else if (
+                objectEvent.message.type == 'membership' &&
+                objectEvent.message.event == 'set' &&
+                objectEvent.message.data.uuid.id == pubnub.getUUID()
+            ) {
+                //  We have joined a channel, logic for this is handled elsewhere.
+                //  No action required
+            }
+            else if (
+                objectEvent.message.type == 'membership' &&
+                objectEvent.message.event == 'set' &&
+                objectEvent.message.data.uuid.id != pubnub.getUUID()
+            ) {
+                //  Somebody else has joined a channel
+                //  Regardless of our active channel, add this person to our list of direct chats if they aren't there already (this is our indication a new user is added)
+                var userId = objectEvent.message.data.uuid.id;
+                if (!channelMembers.hasOwnProperty(userId)) {
+                    //  Find out the information about this user
+                    const userInfo = await getUUIDMetaData(userId);
+
+                    if(!channelMembers.hasOwnProperty(userId)){
+                        if (userInfo != null) {
+                            addUserToCurrentChannel(
+                                userId,
+                                userInfo.data.name,
+                                userInfo.data.profileUrl
+                            )
+                        }
+                    }
+                }
+            }
+            else if (
+                objectEvent.message.type == 'membership' &&
+                objectEvent.message.event == 'delete' &&
+                objectEvent.message.data.uuid == pubnub.getUUID()
```

Comment:
Take a look at what I did recently here, I changed the call to pubnub.getUserId() rather than getUUID().  Strangely, I couldn't get the official getUserId() to work and still need to follow up on that, so I added my own method.  The reason is there was a large effort about a year ago to move away from 'uuid' terminology because technically, the ID provided does not have to be universally unique, just unique to the subkey